File: low-rank-matrix-completion/image-completion/RandomMask.m
function Omega = RandomMask(m,n,k)
%creates a random mxn logical matrix (mask) Omega with exactly k ones
entries = randperm(m*n,k);
Omega_linear = zeros(m*n,1);
Omega_linear(entries) = 1;
Omega = reshape(Omega_linear,m,n);
Omega = Omega==1;
end

File: graduate-research-python/random_mask.py
import numpy as np

def random_mask(m,n,k):
    #creates a random mxn mask with exactly k true entries
    perm = np.random.choice(m*n, k, replace=False)
    Omega_linear = np.zeros(m*n)
    Omega_linear[perm] = 1
    Omega = np.reshape(Omega_linear,(m,n))
    Omega = Omega==1 #turns Omega into a logical matrix
    return Omega